File: src/06-funciones/funciones.py
def capitalizar(nombre):
    capitalizarActual = True
    listaNombre = list(nombre)
    for indice,letra in enumerate(listaNombre):
        if (capitalizarActual and listaNombre[indice] != " "):
            listaNombre[indice] = listaNombre[indice].upper()
            capitalizarActual = False
        if (listaNombre[indice] == " "):
            capitalizarActual = True
    return "".join(listaNombre)

def saludar(nombre):
    print("Hola " + capitalizar(nombre))

if __name__ == "__main__":
    saludar(input("Escribe tu nombre completo: "))

File: src/06-funciones/funciones.pas
Program Funciones;

Function Capitalizar(nombre: String) : String; Forward;
Procedure Saludar(nombre : String);
Begin
    WriteLn('Hola ' + Capitalizar(nombre) + '!');
End;

Function Capitalizar(nombre: String) : String;
Var
capitalizarActual: Boolean;
indice: Integer;
caracterActual: Char;
Begin
    indice := 1;
    capitalizarActual := True;

    For caracterActual In nombre Do
    Begin
        If capitalizarActual and (caracterActual <> ' ') Then
        Begin
            nombre[indice] := Upcase(caracterActual);
            capitalizarActual := False;
        End;

        If caracterActual = ' ' Then
        Begin
            capitalizarActual := True;
        End;

        indice := indice + 1;
    End;

    Capitalizar := nombre;
End;

Var
nombre: String;
Begin
    Write('Ingrese su nombre completo: ');
    ReadLn(nombre);
    Saludar(nombre);
End.
